From 1f209be5604d5e49209349251c604174f7acaa59 Mon Sep 17 00:00:00 2001
From: ming_qian <ming.qian@nxp.com>
Date: Tue, 19 Nov 2019 10:07:20 +0800
Subject: [PATCH 15/37] MLK-22999:VPU Decoder: release debugfs after open to
 avoid memory leak

commit 15bb5a26637e2dee418f499f4a74f4c3435a372d from
https://source.codeaurora.org/external/imx/linux-imx.git

VPU decoder driver create some debug fs,
but don't set the release function,
in open function, it will alloc some memory,
it should be freed when release function is called.
otherwise, memory will be leaked.

Signed-off-by: ming_qian <ming.qian@nxp.com>
Reviewed-by: Zhou Peng <eagle.zhou@nxp.com>
Reviewed-by: Shijie Qin <shijie.qin@nxp.com>
(cherry picked from commit b4ec76046d7a6d9fd45601636f49a5f9fb9989b2)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/vpu_malone/vpu_b0.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/mxc/vpu_malone/vpu_b0.c b/drivers/mxc/vpu_malone/vpu_b0.c
index e050d126c156..88555d12cfd6 100644
--- a/drivers/mxc/vpu_malone/vpu_b0.c
+++ b/drivers/mxc/vpu_malone/vpu_b0.c
@@ -5125,6 +5125,7 @@ static int dbglog_open(struct inode *inode, struct file *filp)
 static struct file_operations dbglog_fops = {
 	.owner = THIS_MODULE,
 	.open = dbglog_open,
+	.release = single_release,
 	.read = seq_read,
 };
 
@@ -5206,6 +5207,7 @@ static int fwlog_open(struct inode *inode, struct file *filp)
 static const struct file_operations fwlog_fops = {
 	.owner = THIS_MODULE,
 	.open = fwlog_open,
+	.release = single_release,
 	.read = seq_read,
 };
 
-- 
2.17.1

